ontology_processing/graph_creation/make_graph_class.py

```diff
@@ -0,0 +1,652 @@
+owlready2.reasoning.JAVA_MEMORY = 500
+
+class MakeGraph:
+    def __init__(self, onto_path, edge_path, output_folder_path="."):
+        self.onto_path = onto_path
+        self.edge_path = edge_path
+        self.output_folder_path = output_folder_path
+        self.onto = None
+        self.object_properties = None
+        self.annot_properties = None
+        self.data_properties = None
+        self.all_myths = None
+        self.G = nx.DiGraph()
+        self.B = None
+        self.B_annotated = None
+        self.subgraph_upstream = None
+        self.superclasses = None
+        self.subgraph_mitigation = None
+
+    def load_ontology(self):
+        """
+        Load the ontology from the OWL file specified by onto_path.
+        """
+        my_world = owlready2.World()
+        self.onto = my_world.get_ontology(self.onto_path).load()
+
+    def set_properties(self):
+        """
+        Format object properties and annotation properties into Python readable names.
+        """
+        self.obj_properties = list(self.onto.object_properties())
+        self.annot_properties = list(self.onto.annotation_properties())
+        self.data_properties = list(self.onto.data_properties())
+        [give_alias(x) for x in self.obj_properties if x.label]
+        [give_alias(x) for x in self.annot_properties if x.label]
+        [give_alias(x) for x in self.data_properties if x.label]
+
+    def automate_reasoning(self):
+        """
+        Run automated reasoning
+        """
+        with self.onto:
+            sync_reasoner()
+
+    def add_edges_to_graph(self):
+        df_edges = pd.read_csv(self.edge_path)
+        for src, tgt, kind in df_edges.values:
+            self.G.add_edge(src, tgt, type=kind, properties=None)
+
+    def build_attributes_dict(self):
+        cm_class = self.onto.search_one(label="climate mind")
+        self.superclasses = list(cm_class.subclasses())
+
+        # get annotation properties for all objects of the ontology (whether node or class)
+        annot_properties = [
+            thing.label[0].replace(":", "_")
+            for thing in list(self.onto.annotation_properties())
+            if thing.label
+        ]
+
+        # get data properties for all objects of the ontology (whether node or class)
+        data_properties = [
+            thing.label[0].replace(" ", "_")
+            for thing in list(self.onto.data_properties())
+            if thing.label
+        ]
+
+        for node in list(self.G.nodes):
+            ontology_node = self.onto.search_one(label=node)
+
+            attributes_dict = {}
+            self.add_basic_info(attributes_dict, ontology_node)
+            self.add_classes(attributes_dict, ontology_node)
+            self.add_properties(attributes_dict, ontology_node, annot_properties, data_properties)
+            self.add_personal_values(attributes_dict)
+            self.add_radical_political(attributes_dict)
+            self.G.add_nodes_from([(node, attributes_dict)])
```

Comment:
This line should probably be commented to say what it’s doing. Something like adding all the attributes from the attributes dictionary to the specific relevant node in the networkx object?

Reply:
Perfect. Got it.